pkg/gardenlet/controller/federatedseed/extensions/shootextensionstatus_control.go

```diff
@@ -0,0 +1,234 @@
+				shootExtensionStatus.Statuses = removeExtensionStatus(shootExtensionStatus.Statuses, kind)
+				return s.patchShootExtensionStatus(ctx, shootExtensionStatus, shootExtensionStatusCopy, kind)
+			}
+		}
+
+		newEntry := gardencorev1alpha1.ExtensionStatus{
+			Kind:    kind,
+			Type:    extensionType,
+			Purpose: purpose,
+			Status:  *currentProviderStatus,
+		}
+
+		// new provider status that has not yet been synced
+		// add the new ExtensionStatus entry
+		if currentExtensionStatus == nil {
+			shootExtensionStatus.Statuses = append(shootExtensionStatus.Statuses, newEntry)
+			return s.patchShootExtensionStatus(ctx, shootExtensionStatus, shootExtensionStatusCopy, kind)
+		}
+
+		// at this point we know that we have a provider status that also exists in the ShootExtensionStatus in the Garden cluster
+		// we need to check if the status is up to date by checking for equality
+		if apiequality.Semantic.DeepEqual(*currentExtensionStatus, newEntry) {
+			s.log.Debugf("Skipping sync of the ShootExtensionStatus for the %q extension of Shoot %q. The resource is up-to-date", kind, shoot.Name)
+			return reconcile.Result{}, nil
+		}
+
+		// patch the existing ExtensionStatus entry and patch the ShootExtensionStatus resource
+		shootExtensionStatus.Statuses[indexOldStatus] = newEntry
+		return s.patchShootExtensionStatus(ctx, shootExtensionStatus, shootExtensionStatusCopy, kind)
+	}
+}
+
+// getShootForRequest returns the Shoot resource for a reconcile.Request parsed from the Cluster resource in the Seed
+func (s *ShootExtensionStatusControl) getShootForRequest(ctx context.Context, req reconcile.Request) (*gardencorev1beta1.Shoot, error) {
+	cluster, err := extensions.ClusterFromRequest(ctx, s.seedClient, req)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, err
+		}
+		return nil, fmt.Errorf("could not get cluster with name %s : %w", cluster.Name, err)
+	}
+
+	shoot, err := extensions.ShootFromCluster(s.decoder, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	if shoot == nil {
+		return nil, fmt.Errorf("cluster resource %s doesn't contain shoot resource", cluster.Name)
+	}
+	return shoot, nil
+}
+
+func (s *ShootExtensionStatusControl) patchShootExtensionStatus(ctx context.Context, shootExtensionStatus *gardencorev1alpha1.ShootExtensionStatus, shootExtensionStatusCopy *gardencorev1alpha1.ShootExtensionStatus, kind string) (reconcile.Result, error) {
+	if err := s.gardenClient.Patch(ctx, shootExtensionStatus, client.MergeFromWithOptions(shootExtensionStatusCopy, client.MergeFromWithOptimisticLock{})); err != nil {
+		message := fmt.Sprintf("The %q extension status for Shoot %q was NOT successfully synced: %v", shootExtensionStatus.Name, kind, err)
+		s.log.Error(message)
+		return reconcile.Result{}, err
+	}
+
+	message := fmt.Sprintf("The provider status of the %q extension of Shoot %q was successfully synced to the Garden cluster", kind, shootExtensionStatus.Name)
+	s.log.Info(message)
```

Comment:
```suggestion
	s.log.Debug(message)
```

Reply:
The ShootState controller also logs with [Info level on success](https://github.com/gardener/gardener/blob/3d5f3f3fd3d65b57f775cb14fb251463ace8b731/pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go#L135).  Is there a reason you would like to change that?

Reply:
Well, it's just annoying to log such messages on `Info` and unnecessarily spams the logs 😄